### PR TITLE
✨ feat: add a SubRegistry for UniswapV2 Positions 

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/AssetRegistry/UniswapV2PricingModule.sol [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        internal
+        view
+        returns (uint256 reserve0, uint256 reserve1)
+    {
+        // The untrusted reserves from the pair, these can be manipulated!!!
+        (reserve0, reserve1,) = IUniswapV2Pair(pair).getReserves();
+
+        require(reserve0 > 0 && reserve1 > 0, "UV2_GV: ZERO_PAIR_RESERVES");
+
+        // Compute how much to swap to balance the pool with externally observed trusted prices
```
Comment (zekiblue) [inline src/AssetRegistry/UniswapV2PricingModule.sol:243]: Is it possible to early terminate if the pools are balanced

Comment (j-vp) [reply src/AssetRegistry/UniswapV2PricingModule.sol:243]: it's going to be hard to be exactly balanced, the oracle prices of the underlying assets will never be exactly the ratio of the reserves

File: src/interfaces/IUniswapV2Factory.sol [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * This is a private, unpublished repository.
```
Comment (j-vp) [inline src/interfaces/IUniswapV2Factory.sol:2]: to be changed, to MIT

File: src/interfaces/IUniswapV2Pair.sol [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * This is a private, unpublished repository.
+ * All rights reserved to Arcadia Finance.
+ * Any modification, publication, reproduction, commercialization, incorporation,
+ * sharing or any other kind of use of any part of this code or derivatives thereof is not allowed.
+ *
+ * SPDX-License-Identifier: UNLICENSED
```
Comment (j-vp) [inline src/interfaces/IUniswapV2Pair.sol:7]: to be changed, to MIT

File: src/mockups/UniswapV2FactoryMock.sol [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/**
+ * This is a private, unpublished repository.
+ * All rights reserved to Arcadia Finance.
+ * Any modification, publication, reproduction, commercialization, incorporation,
+ * sharing or any other kind of use of any part of this code or derivatives thereof is not allowed.
+ *
+ * SPDX-License-Identifier: UNLICENSED
```
Comment (j-vp) [inline src/mockups/UniswapV2FactoryMock.sol:7]: to be changed, to MIT/BUSL

File: src/mockups/UniswapV2PairMock.sol [hunk 5]
```diff
@@ -0,0 +1,115 @@
+/**
+ * This is a private, unpublished repository.
+ * All rights reserved to Arcadia Finance.
+ * Any modification, publication, reproduction, commercialization, incorporation,
+ * sharing or any other kind of use of any part of this code or derivatives thereof is not allowed.
+ *
+ * SPDX-License-Identifier: UNLICENSED
```
Comment (j-vp) [inline src/mockups/UniswapV2PairMock.sol:7]: to be changed, to MIT

File: src/test/12_UniswapV2PricingModule.t.sol [hunk 6]
```diff
@@ -0,0 +1,897 @@
...
+        );
+        mainRegistry.addPricingModule(address(uniswapV2PricingModule));
+        vm.stopPrank();
+    }
+
+    //this is a before each
+    function setUp() public virtual {}
+}
+
+contract OldTests is UniswapV2PricingModuleTest {
```
Comment (j-vp) [inline src/test/12_UniswapV2PricingModule.t.sol:237]: still need those?

File: src/test/12_UniswapV2PricingModule.t.sol [hunk 7]
```diff
@@ -0,0 +1,897 @@
...
+
+        //Then: pairSnxEth is not white-listed
+        assertTrue(!uniswapV2PricingModule.isWhiteListed(randomAsset, 0));
+    }
+}
+
+/*///////////////////////////////////////////////////////////////
+                        PRICING LOGIC
+///////////////////////////////////////////////////////////////*/
+contract PricingLogic is UniswapV2PricingModuleTest {
```
Comment (j-vp) [inline src/test/12_UniswapV2PricingModule.t.sol:828]: extra pricing tests todo

**Review by j-vp — COMMENTED** [review]
to main or to dev branch?

**Pull request merged**